astropy/units/structured.py

```diff
@@ -0,0 +1,480 @@
+            # Index with empty tuple to decay array scalars to numpy void.
+            return result if result.shape else result[()]
+
+        return converter
+
+    def to(self, other, value=np._NoValue, equivalencies=[]):
+        """Return values converted to the specified unit.
+
+        Parameters
+        ----------
+        other : `~astropy.units.StructuredUnit`
+            The unit to convert to.  If necessary, will be converted to
+            a `~astropy.units.StructuredUnit` using the dtype of ``value``.
+        value : array-like, optional
+            Value(s) in the current unit to be converted to the
+            specified unit.  If a sequence, the first element must have
+            entries of the correct type to represent all elements (i.e.,
+            not have, e.g., a ``float`` where other elements have ``complex``).
+            If not given, assumed to have 1. in all fields.
+        equivalencies : list of tuple, optional
+            A list of equivalence pairs to try if the units are not
+            directly convertible.  See :ref:`unit_equivalencies`.
+            This list is in addition to possible global defaults set by, e.g.,
+            `set_enabled_equivalencies`.
+            Use `None` to turn off all equivalencies.
+
+        Returns
+        -------
+        values : scalar or array
+            Converted value(s).
+
+        Raises
+        ------
+        UnitsError
+            If units are inconsistent
+        """
+        if value is np._NoValue:
+            # We do not have UNITY as a default, since then the docstring
+            # would list 1.0 as default, yet one could not pass that in.
+            value = UNITY
+        return self._get_converter(other, equivalencies=equivalencies)(value)
+
+    def to_string(self, format='generic'):
+        """Output the unit in the given format as a string.
+
+        Units are separated by commas.
+
+        Parameters
+        ----------
+        format : `astropy.units.format.Base` instance or str
+            The name of a format or a formatter object.  If not
+            provided, defaults to the generic format.
+
+        Notes
+        -----
+        Structured units can be written to all formats, but can be
+        re-read only with 'generic'.
+
+        """
+        parts = [part.to_string(format) for part in self.values()]
+        out_fmt = '({})' if len(self) > 1 else '({},)'
+        if format == 'latex':
+            # Strip $ from parts and add them on the outside.
+            parts = [part[1:-1] for part in parts]
+            out_fmt = '$' + out_fmt + '$'
+        return out_fmt.format(', '.join(parts))
+
+    def _repr_latex_(self):
+        return self.to_string('latex')
+
+    __array_ufunc__ = None
+
+    def __mul__(self, other):
+        if isinstance(other, str):
+            try:
+                other = Unit(other, parse_strict='silent')
+            except Exception:
+                return NotImplemented
```

Comment:
When does this arise? When other is not a unit string?
And can the error be specified, like UnitConversionError or something?

Reply:
Yes, when `other` is something arbitrary. Here, the goal is simply to check whether the structured unit knows how to deal with other, and I think catching arbitrary exceptions is correct - we should definitely return `NotImplemented` on anything that goes wrong, because that means we do not know how to deal with it, and `other` should get a chance.